crypto/crypto.go

```diff
@@ -16,23 +23,26 @@ type Encrypted struct {
 
 type Signed struct {
 	Message string
-	Mode    string
+	Mode    Mode
 	//Inputs map[string]string
 	Signature string
 }
 
-func NewSignature() *Signed {
-	return &Signed{Mode: SignatureMode}
-}
-
 func NewHMAC() *Signed {
 	return &Signed{Mode: HMACMode}
 }
 
+func NewSignature(mode Mode) *Signed {
+	return &Signed{Mode: mode}
+}
+
 func GroupEncrypt(plaintext string, publicKeys map[string]string) (*Encrypted, error) {
 
 	keySize := 32
-	key := RandomBytes(keySize)
+	key, err := RandomBytes(keySize)
+	if err != nil {
+		return nil, err
```

Comment:
I think library code (which this package seems to be) should explicitly return errors instead of panicing. I changed the interface of some functions in crypto/helpers.go to do this. RandomBytes() was one of those function. If I recall, the error returned from RandomBytes() already adds some context to the error, which is why I simply returned the error here. If prefered , we can add even more context.